plugins/scanners/fossid/src/main/kotlin/FossId.kt

```diff
@@ -276,9 +276,8 @@ class FossId internal constructor(
                         "path '${pkg.vcsProcessed.path}' set and scanning cannot be limited to paths.",
                     severity = Severity.WARNING
                 )
-                val provenance = RepositoryProvenance(pkg.vcsProcessed, pkg.vcsProcessed.revision)
 
-                return createSingleIssueResult(issue, provenance)
+                return createSingleIssueResult(issue, nestedProvenance?.root ?: UnknownProvenance)
```

Comment:
Thanks for the hints, I've done all of it except for

> In this context it would also make sense to change the if conditions to work with `provenance` instead of `pkg.vcsProcessed`.

I'm running into a chicken-and-egg problem when using the provenance for error checking as I cannot construct a provenance with a blank revision. I'll think about this a bit more.